mods/ra/maps/siberian-conflict-1-fresh-tracks/fresh-tracks.lua

```diff
@@ -0,0 +1,282 @@
+	end
+end
+
+FinalTrucks = function()
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Media.PlaySpeechNotification(Allies, "ConvoyApproaching")
+		local basePath = BaseConvoyPath
+		local baseWaypoint = basePath[#basePath]
+		local baseConvoy = Reinforcements.Reinforce(USSR, BaseConvoyUnits, { basePath[1] })
+		Utils.Do(baseConvoy, function(unit)
+			Trigger.OnAddedToWorld(unit, function()
+				Utils.Do(basePath, function(waypoint)
+					unit.Move(waypoint)
+				end)
+
+				Trigger.OnIdle(unit, function()
+					unit.Move(baseWaypoint)
+				end)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(2), function()
+		Media.PlaySpeechNotification(Allies, "ConvoyApproaching")
+		local finalPath = FinalConvoyPath
+		local finalWaypoint = finalPath[#finalPath]
+		local finalConvoy = Reinforcements.Reinforce(USSR, FinalConvoyUnits, { finalPath[1] })
+		Utils.Do(finalConvoy, function(unit)
+			Trigger.OnAddedToWorld(unit, function()
+				Utils.Do(finalPath, function(waypoint)
+					unit.Move(waypoint)
+				end)
+
+				Trigger.OnIdle(unit, function()
+					unit.Move(finalWaypoint)
+				end)
+			end)
+		end)
+
+		Trigger.OnAllKilled(Utils.Where(finalConvoy, function(a) return a.Type == "truk" end), function()
+			Allies.MarkCompletedObjective(StopTrucks)
+		end)
+	end)
+end
+
+ConvoyExit = function()
+	Trigger.OnEnteredFootprint(ConvoyEscape, function(actor, id)
+		if actor.Owner == USSR and actor.Type == "truk" then
+			actor.Stop()
+			actor.Destroy()
+			Allies.MarkFailedObjective(StopTrucks)
+		elseif actor.Owner == USSR and actor.Type ~= "truk" then
+			actor.Stop()
+			actor.Destroy()
+		end
+	end)
+end
+
+BridgeTriggers = function()
+	Trigger.OnKilled(BridgeBarrel1, function()
+		local theNorthEastBridge = Utils.Where(Map.ActorsInWorld, function(actor) return actor.Type == "bridge2" end)[1]
+		if not theNorthEastBridge.IsDead then
+			theNorthEastBridge.Kill()
+		end
+	end)
+
+	Trigger.OnKilled(BridgeBarrel2, function()
+		local theNorthwestBridge = Map.ActorsInBox(NorthwestPath1.CenterPosition, NorthEntry.CenterPosition, function(self) return self.Type == "bridge1" end)[1]
+		if not theNorthwestBridge.IsDead then
+			theNorthwestBridge.Kill()
+		end
+	end)
+
+	Trigger.OnKilled(BridgeBarrel3, function()
+		local theSoutheastBridge = Map.ActorsInBox(SoutheastBridge.CenterPosition, SouthPath3.CenterPosition, function(self) return self.Type == "bridge1" end)[1]
+		if not theSoutheastBridge.IsDead then
+			theSoutheastBridge.Kill()
+		end
+	end)
+
+	AllThreeBridges = Utils.Where(Map.ActorsInWorld, function(actor)
+		return
+			actor.Type == "bridge1" or
+			actor.Type == "bridge2"
+	end)
+
+	Trigger.OnAllKilled(AllThreeBridges, function()
+		Allies.MarkCompletedObjective(DestroyBridges)
+	end)
+end
+
+FinishTimer = function()
+	for i = 0, 5 do
+		local c = TimerColor
+		if i % 2 == 0 then
+			c = HSLColor.White
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(i), function() UserInterface.SetMissionText("Here they come!", c) end)
```

Comment:
Imo it would be better to have a more formal tone in the UI. What about just "The first convoy has entered the area"?

Reply:
How about: "The first trucks are entering your AO."

Reply:
That should be fine.